### PR TITLE
[cirque] Add cirque test for Python device controller

### LINKED ISSUE #4645: Add e2e cirque test between python device controller and lighting example regarding thread network provisioning
#### Problem
Add e2e cirque test between python device controller and lighting example regarding thread network provisioning

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem

We are missing some tests for Python binding of chip device controller, this test adds them.

 #### Summary of Changes

1. Add cirque test for python device controller.
2. Update log outputs for tests
3. Wrap tests using `catchsegv` to catch potential crashes.

 Fixes #4645

### CODE REVIEW

**Comment by erjiaqing** [comment]
@gjc13 Seems ThreadStackMgr on Linux lighting app does not send the message successfully, do you have any ideas about this?

**Review by andy31415 — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Comment by erjiaqing** [comment]
@jelderton  @LuDuda @saurabhst PTAL, thanks

File: src/app/clusters/network-commissioning/network-commissioning.cpp [hunk 1]
```diff
@@ -217,7 +217,11 @@ CHIP_ERROR DoEnableNetwork(NetworkInfo * network)
     {
     case NetworkType::kThread:
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+// TODO: On linux, we are using Reset() instead of Detach() to disable thread network, which is not expected.
+// Upstream issue: https://github.com/openthread/ot-br-posix/issues/755
+#if !CHIP_DEVICE_LAYER_TARGET_LINUX
         ReturnErrorOnFailure(DeviceLayer::ThreadStackMgr().SetThreadEnabled(false));
```
Comment (Damian-Nordic) [inline src/app/clusters/network-commissioning/network-commissioning.cpp:223]: If the behaviour is platform-dependent, maybe we could call `SetThreadEnabled()` as part of `SetThreadProvision()`?

File: src/controller/python/test/mobile-device-test.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        try:
+            self.devCtrl.ZCLSend("NetworkCommissioning", "EnableNetwork", nodeid, 1, 0, {
+                "networkID": bytes.fromhex(TEST_THREAD_NETWORK_ID),
+                "breadcrumb": 0,
+                "timeoutMs": 1000})
+        except Exception as ex:
+            self.logger.exception("Failed to send EnableNetwork command")
+            return False
+        # Wait for thread network join
+        time.sleep(45)
```
Comment (LuDuda) [inline src/controller/python/test/mobile-device-test.py:108]: nit: What is the reason to wait 45 seconds for joining, it seems too much.

**Review by LuDuda — APPROVED** [review]
This is great! Thanks :+1:

File: src/controller/python/test/mobile-device-test.py (hunk 2 above)
Comment (andy31415) [reply src/controller/python/test/mobile-device-test.py:108]: Generally we should avoid all sleeps. Could we poll for some sort status change or away a 'joined' notification?

Comment (erjiaqing) [reply src/controller/python/test/mobile-device-test.py:108]: That kind of implementation will be added in later IM & cluster updates. For now I think we can set it to 15 seconds, which is the time we used in SetupThreadNetwork in Cirque tests.

Comment (erjiaqing) [reply src/controller/python/test/mobile-device-test.py]: Added a comment for which feature is required before we can waiting for network join in a CHIP way.

File: src/app/clusters/network-commissioning/network-commissioning.cpp (hunk 1 above)
Comment (erjiaqing) [reply src/app/clusters/network-commissioning/network-commissioning.cpp:223]: As described in the comments, this special check will be removed after upstream change of ot-br-posix.
However, I think we can introduce a helper function maybe named `ThreadStackMgr::AttachThreadNetwork`?

@bukepo Do you have any idea about this?

Anyway, this is not the topic of this PR -- it just add a test and make minimal changes to existing code to make it available.

Comment (bukepo) [reply src/app/clusters/network-commissioning/network-commissioning.cpp:223]: Can we change the behavior of Reset in ot-br-posix to just do network reset?

**Comment by erjiaqing** [comment]
@Damian-Nordic your comment will be fixed after upstream change is applied, PTAL.

**Comment by mspang** [comment]
Please don't use sleep(1) for synchronizing tests. We don't want flaky tests, e.g. if GitHub CI is under heavy load.

**Pull request merged**